gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageWriteChannel.java

```diff
@@ -20,11 +20,11 @@
 @ThreadSafe
 final class CloudStorageWriteChannel implements SeekableByteChannel {
 
-  private GcsOutputChannel channel;
+  private WriteChannel channel;
```

Comment:
Truncate is how I thought the GCS API works by default. I'm not sure if this code will work if you open a file for writing, without the truncate flag, and try to read its contents. I need to have a closer look at how the veneer library works. I notice that it's using gRPC calls that look a lot like a `write` system call. Are there any good docs that talk about the technical details of how this cloud storage api works?